flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -258,12 +270,28 @@ public void addAll(Configuration other, String prefix) {
         bld.append(prefix);
         final int pl = bld.length();
 
-        synchronized (this.confData) {
-            synchronized (other.confData) {
-                for (Map.Entry<String, Object> entry : other.confData.entrySet()) {
-                    bld.setLength(pl);
-                    bld.append(entry.getKey());
-                    this.confData.put(bld.toString(), entry.getValue());
+        Object lock1 = this.confData;
```

Comment:
I suggest lock1 -> thisConfDataLock
lock2 -> otherConfDataLock

Reply:
How about `thisLock` and `otherLock` ?